tests/unit/preprocessor/_time/test_time.py

```diff
@@ -905,82 +905,87 @@ def make_map_data(number_years=2):
     return cube
 
 
-@pytest.mark.parametrize('standardize', [False, True])
-@pytest.mark.parametrize('period', ['full', 'day', 'month', 'season'])
-def test_anomalies(period, standardize):
-    cube = make_map_data(number_years=2)
+@pytest.mark.parametrize('standardize', [True])
+@pytest.mark.parametrize('period', ['full'])
+def test_standardized_anomalies(period, standardize):
+    cube = make_map_data(number_years=4)
     result = anomalies(cube, period, standardize)
     if period == 'full':
         expected_anomalies = (cube.data - np.mean(cube.data, axis=2, keepdims=True))
         if standardize:
-            # TODO: add numpy array with correct values here
+            # NB: default numpy behaviour is ddof=0,
+            # whereas default iris behaviour is ddof=1
             expected_stdanomalies = expected_anomalies / np.std(
-                 expected_anomalies, axis=2, keepdims=True)
+                 expected_anomalies, axis=2, keepdims=True, ddof=1)
             expected = np.ma.masked_invalid(expected_stdanomalies)
-            assert_allclose(
+            assert_array_equal(
                 result.data,
-                expected,
-                # Note the pretty large tolerance, which is needed since
-                # there actually IS quite a difference --> WHY?
-                rtol=1.2e-5,
-                atol=0.0012
+                expected
```

Comment:
you can still use `assert_allcolse` with an `rtol = 1e-5` like in https://numpy.org/doc/1.18/reference/generated/numpy.testing.assert_allclose.html

Reply:
yes, but we don't need it anymore, not for this case at least.  It is better to use `assert_array_equal`, see latest commits. :)  I think this is nice, imagine in the future Iris decides (for whatever reason) to change `ddof` to 0, this test will complain :) and it should, since something is actually changing the results that would go unnoticed otherwise.

Reply:
actually, it's better to use `allclose` with a say, `1e-5` tolerance since if iris changes and the test fails it means that the results differ by more than 1e-5 which is pretty bad (1e-4 for a temperature filed is already kinda bad, that means variations of order 0.01K per point which is edging on dangerous)

Reply:
anyways, I might sound like Dr Pedanticus :grin:

Reply:
I don't get your point, the testing as I currently implemented is more strict than what you are suggesting.

Reply:
dont worry about it - all good as is now - my point was if the tolerances change for the iris operation this test as is now will fail, but if the abs diffs are still within 1e-5 that's fine, hence using `allclose` will still let the test pass. But it's fine, since if it fails, that'll prompt us to re evaluate the new differences anyway and see what's happened :+1:

Reply:
> if it fails, that'll prompt us to re evaluate the new differences anyway and see what's happened
That was exactly my intention, this test should really flag when something changes. If we would have a very clear view on tolerable errors, we could implement that, but I would feel uncomfortable setting a threshold here.